### ISSUE TITLE
Is it normal that gitpod is slow on first time load in browser?

### ISSUE THREAD

**Issue by ekilzen** [gitpod-io/gitpod#560]
It seems that gitpod is very slow on first time opening it on new browser, estimated to take 5 minutes and more to load up. After it cached, it works fine.  Is it normal situation?

**Comment by alesanchezr** [comment]
It is for me, the next time you open the repository is faster, even if you start a new container from scratch.

It also depends on how custom made is your container, I believe they have cached snapshots or images for the most popular configurations.

**Comment by svenefftinge** [comment]
5 minutes is definitely too long!

But, yes, the IDE is quite big and needs to be downloaded first time.

The starting takes time depending on
 - whether the docker image is readily built and already pulled to the local machine.
 - size of git repository / resp. size of snapshoted workspace

We have measurements in place and regularly look into it. We've seen outliers as you mentioned and trying to understand what could cause them. Generally startup time is crucial and we will probably never stop investing into making things start faster.

**Comment by lastingman** [comment]
Thanks for explain it to me. The idea of Gitpod is great and I am enjoy using it. 

Is the load up speed depend on the region? In my situation, I am in china, and I am using VPN which server is located in HongKong.

If this kind of side effect can be reduced, I think it will benefit lots of people.

**Comment by meysholdt** [comment]
I measured the times of the request with Chrome Dev tools when opening [apache-example](https://github.com/gitpod-io/apache-example). The measurements do not include a prebuild or an image build. I did clear my local cache before I measured this, so the measurements should be representative for a first-time user.

all requests chronologically:
![image](https://user-images.githubusercontent.com/239422/57508610-37836e80-7302-11e9-811d-118fd5bc5bbf.png)

all requests sorted by file size:
![image](https://user-images.githubusercontent.com/239422/57508636-4cf89880-7302-11e9-9487-90af66986ceb.png)
